### ISSUE TITLE
php 7.2 method / constructor notice

### ISSUE THREAD

**Issue by jools-r** [Bloke/smd_xml#1]
With php 7.2, the following notice is given in debugging mode:

```
Methods with the same name as their class will not be constructors in a future version of PHP; smd_xml_build_data has a deprecated constructor on line 804
```
referring to [this line](https://github.com/Bloke/smd_xml/blob/master/smd_xml.php#L802).

You can probably spot it a mile off, but I had to read up. The solution is apparently as simple as renaming the function to `__construct`, i.e.:

```
function __construct ($data, $startElement, $recWrap, $xml_version = '1.0', $xml_encoding = 'UTF-8')
```

Does that look feasible to you?

**Comment by Bloke** [comment]
Yep, been on my mental list for ages. I think smd_calendar suffers the same fate. Thanks for the reminder.

**Comment by jools-r** [comment]
Fixed in da170e48c808073bf7aabdc917cfdcb58296e795
